utils/threshold.go

```diff
@@ -49,8 +49,16 @@ func (m Threshold) LTE(t Threshold) bool {
 
 // Validate returns an error if threshold is invalid
 func (m Threshold) Validate() error {
-	if m.Denominator == 0 {
-		return fmt.Errorf("denominator must not be 0")
+	if m.Numerator <= 0 {
+		return fmt.Errorf("threshold numerator must be a positive integer")
+	}
```

Comment:
do we really need this check? Why shouldn't we allow a threshold that is basically 0?

Reply:
Don't think we ever have that requirement.

Reply:
None of our thresholds should be zero. And I also want a sanity check against default values.